### PR TITLE
tidb: sleep for a while and retry when schema out of date

### PR DESCRIPTION
**Pull request by tiancaiamao**
also adjust schema lease duration in ddl test

### CODE REVIEW

File: session.go [hunk 1]
```diff
@@ -151,10 +151,26 @@ func (s *session) SetConnectionID(connectionID uint64) {
 
 type schemaLeaseChecker struct {
 	domain.SchemaValidator
-	schemaVer int64
+	schemaVer     int64
+	retryInterval time.Duration
 }
 
 func (s *schemaLeaseChecker) Check(txnTS uint64) error {
+	for i := 0; i < 5; i++ {
```
Comment (shenli) [inline session.go]: How many time will it retry in the old implementation?

File: session.go [hunk 2]
```diff
@@ -192,6 +208,7 @@ func (s *session) doCommit() error {
 	s.txn.SetOption(kv.SchemaLeaseChecker, &schemaLeaseChecker{
 		SchemaValidator: sessionctx.GetDomain(s).SchemaValidator,
 		schemaVer:       s.sessionVars.TxnCtx.SchemaVersion,
+		retryInterval:   sessionctx.GetDomain(s).DDL().GetLease(),
```
Comment (shenli) [inline session.go]: Sleeping for a lease is too long.

**Comment by shenli** [comment]
LGTM @coocood @zimulala PTAL

File: session.go [hunk 3]
```diff
@@ -154,7 +154,24 @@ type schemaLeaseChecker struct {
 	schemaVer int64
 }
 
+const schemaOutOfDateRetryInterval = 500 * time.Millisecond
+
 func (s *schemaLeaseChecker) Check(txnTS uint64) error {
+	for i := 0; i < 10; i++ {
```
Comment (coocood) [inline session.go]: extract 10 as constant.

**Comment by coocood** [comment]
LGTM

**Pull request merged**